code/modules/reagents/chemistry/holder.dm

```diff
@@ -653,39 +654,52 @@
 	var/list/cached_reagents = reagent_list
 	if(owner)
 		expose_temperature(owner.bodytemperature, 0.25)
-	var/need_mob_update = 0
+	var/need_mob_update = FALSE
 	for(var/datum/reagent/reagent as anything in cached_reagents)
-		if(QDELETED(reagent.holder))
-			continue
-
-		if(!owner)
-			owner = reagent.holder.my_atom
-
-		if(owner && reagent)
-			if(owner.reagent_check(reagent, delta_time, times_fired) != TRUE)
-				if(liverless && !reagent.self_consuming) //need to be metabolized
-					continue
-				if(!reagent.metabolizing)
-					reagent.metabolizing = TRUE
-					reagent.on_mob_metabolize(owner)
-				if(can_overdose)
-					if(reagent.overdose_threshold)
-						if(reagent.volume >= reagent.overdose_threshold && !reagent.overdosed)
-							reagent.overdosed = TRUE
-							need_mob_update += reagent.overdose_start(owner)
-							log_game("[key_name(owner)] has started overdosing on [reagent.name] at [reagent.volume] units.")
-					for(var/addiction in reagent.addiction_types)
-						owner.mind?.add_addiction_points(addiction, reagent.addiction_types[addiction] * REAGENTS_METABOLISM)
-
-					if(reagent.overdosed)
-						need_mob_update += reagent.overdose_process(owner, delta_time, times_fired)
-
-				need_mob_update += reagent.on_mob_life(owner, delta_time, times_fired)
+		need_mob_update += metabolize_reagent(owner, reagent, delta_time, times_fired, can_overdose, liverless)
 	if(owner && need_mob_update) //some of the metabolized reagents had effects on the mob that requires some updates.
 		owner.updatehealth()
 		owner.update_stamina()
 	update_total()
 
+/*
+ * Metabolises a single reagent for a target owner carbon mob. See above.
+ *
+ * Arguments:
+ * * mob/living/carbon/C - The mob to metabolize in, if null it uses [/datum/reagents/var/my_atom]
```

Comment:
```suggestion
 * * mob/living/carbon/owner - The mob to metabolize in, if null it uses [/datum/reagents/var/my_atom]
```

Reply:
Also, you have other arguments, you should document those too.

Reply:
I _think_ this is what they are?

Reply:
Yeah I looked it up and corrected it